yb-voyager/cmd/exportSchema.go

```diff
@@ -195,24 +206,85 @@ func exportSchema() error {
 	return nil
 }
 
-func nudgeUserToRunAssessment() error {
-	msr, err := metaDB.GetMigrationStatusRecord()
-	if err != nil {
-		return fmt.Errorf("failed to get migration status record: %w", err)
+func runAssessMigrationCmdBeforExportSchemaIfRequired(exportSchemaCmd *cobra.Command) error {
+	if source.DBType != POSTGRESQL {
+		log.Infof("skipping running assess-migration command from export schema as source DB type is not PostgreSQL.")
+		return nil
 	}
 
-	if msr.MigrationAssessmentDone {
+	if ok, _ := IsMigrationAssessmentDoneDirectly(metaDB); ok {
+		log.Infof("migration assessment is already done, skipping running assess-migration command.")
 		return nil
-	}
-	if source.DBType != POSTGRESQL {
-		// At this point, we are only interested in nudging postgres users to run assessment
-		// because it is far more advanced and useful for them.
+	} else if ok, _ := IsMigrationAssessmentDoneViaExportSchema(); ok {
+		log.Infof("migration assessment is already done via export schema, skipping running assess-migration command.")
 		return nil
 	}
-	if !utils.AskPrompt("It is recommended to run assess-migration before exporting schema. https://docs.yugabyte.com/preview/yugabyte-voyager/migrate/assess-migration/ \n" +
-		"Do you want to continue anyway without running assess-migration") {
-		utils.ErrExit("Aborting...")
+
+	// Temporarily override utils.ErrExit with panic so it doesn’t kill the export schema command.
+	utils.SetExitHook(func(code int) { panic(fmt.Sprintf("exit %d", code)) })
+	defer utils.SetExitHook(nil) // restore when we’re done
+
+	// Note: have to maintain it in metaDB, can't rely on in-memory var to cover multiple runs scenario
+	isAssessmentInvokedFromExportSchema = true
+
+	var assessFlagsWithValues []string
+	commonFlags := utils.GetCommonFlags(exportSchemaCmd, assessMigrationCmd)
+	for _, flag := range commonFlags {
+		// don't pass start-clean flag to assess-migration command here
+		if flag.Name == "start-clean" || !flag.Changed {
+			continue
+		}
+
+		// bool flags: --flag=value
+		// everything else: --flag value
+		if flag.Value.Type() == "bool" {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				fmt.Sprintf("--%s=%s", flag.Name, flag.Value.String()),
+			)
+		} else {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				"--"+flag.Name,
+				flag.Value.String(),
+			)
+		}
 	}
+
+	// Append --yes=true(irrespective) at the end to override any --yes=false if set in export schema cmd
+	assessFlagsWithValues = append(assessFlagsWithValues,
+		"--iops-capture-interval", "0", // TODO: any small but significant duration will be better than 0
+		"--target-db-version", ybversion.LatestStable.String(),
+		"--yes", // need to be restored
+	)
+
+	// snapshot the global variable DoNotPrompt to restore it later
+	prevDoNotPrompt := utils.DoNotPrompt
```

Comment:
I still feel the separate process is cleaner and better 😅 , looks like any global variable or parameter common across the commands we need the handling like this. So in the future, whenever we add new things we need to take care of that else it could lead incorrect behaviour.

- It won't affect the current export-schema defaults or any common parameters or variables.
- We can handle the errors directly so the err-exit handling is also cleaner that way.

cc: @makalaaneesh @sanyamsinghal ?

Reply:
> looks like any global variable or parameter common across the commands we need the handling like this

Yes, if we are setting any for assess specifically.

Reply:
I think I agree with Priyanshi. @sanyamsinghal what were the challenges with invoking a separate command? (I think we discussed this but don't recall 😅 )
Separate command is not clean for sure, but it seems like it would be cleaner than the current approach (where we have to mess around with a lot of variables and reset the state later.

Reply:
will it cause problems in docker environment?

Reply:
> what were the challenges with invoking a separate command?

Tracking whether the assessment was done explicitly or via export schema cmd.